### ISSUE TITLE
Disabling mouse input prevents hiding the mouse

### ISSUE THREAD

**Issue by marat569** [SpecialKO/SpecialK#15]
**Special K version**: 22.7.19

**Game**: Any compatible game/api

**Problem**: If enabling "Disable Mouse Input to Game" along with "Hide When Not Moved", the default windows pointer will either stay on screen or just "park" itself in the middle. Apparently the functionality worked fine before, but "broke" around the time Replicant was released. 

The cursor sticks out like a sore thumb, and sadly makes the "Disable Mouse Input" option unusable. 

**How to replicate**: 
1) Enable "Disable Mouse Input to game"

2) Enable "Hide When Not Moved" (Set to 0.00 for instant results)

![image](https://user-images.githubusercontent.com/7017110/180598258-37dc78c9-689b-4a1c-bed2-4e73d1ee4136.png)

**Desired results**: When enabling "Hide When Not Moved" along with "Disable Mouse Input", no mouse cursor would be visible in the game window; unless opening the SK GUI.

**Comment by Aemony** [comment]
There's mainly two different ways of solving this as I see it:

* Fix whatever bug is preventing the use of "Hide When Not Moved" when "Disable Mouse Input to Game" is enabled.
* **Always** hide the mouse cursor automatically when "Disable Mouse Input to Game" is enabled.
  * An alternative here is to only hide the mouse cursor when the game has focus and it is within the game window.

Hiding the mouse automatically when disabling mouse input might not be desirable in multi-monitor setups where the mouse is constantly moved across the game window, as that might cause the user to lose track of the mouse cursor in the transit. Though if it's possible to only hide the cursor when the game has focus and it's being moved across the game window that would probably be the best solution.

**Comment by Kaldaien** [comment]
The bug that prevents this from working has to do with the way idle cursor is detected. When the block mouse input to game option is selected, that also never runs the hook code that detects cursor motion.

I've known about this for years, but I don't really know a good solution. The current solution is actually intentional, since the hook code never executes the cursor would be hidden constantly and potentially make SK's UI unusable.

**Comment by marat569** [comment]
Maybe make a "Band-Aid" hidden setting that is only available in the .ini.

For example I'm done changing all the settings in a game, don't really need SKGUI anymore; so I can like toggle the setting and hide the mouse; at the cost of no more SKUI

**Comment by Aemony** [comment]
This have now been fixed and is included in v 22.8.4.3 or later.

```
22.8.4.3-D3D9-IK
================
 + Added support for mouse cursor auto-hide when using 'Block Mouse Input to Game'
 + Fixed incorrect mouse activation behavior when background render is enabled and
     the game window is activated by clicking on SK's UI
 + Remove spurious calls to glClampColor (...) to clamp vertex/fragment colors,
     that feature is deprecated in GL3 core profiles
```

Download link for v 22.8.4.3: https://sk-data.special-k.info/repository/SpecialK_22.8.4.3.exe
